framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -777,8 +783,14 @@ private Range calculateRangeUnaryOp(
             TransferInput<CFValue, CFStore> p,
             CFStore thenStore,
             CFStore elseStore) {
-        if (isIntRange(leftNode, p) || isIntRange(rightNode, p)) {
-            return refineIntRanges(leftNode, rightNode, op, p, thenStore, elseStore);
+        if (isIntRange(leftNode, p)
```

Comment:
I think there should be a comment here explaining why the check against UnknownVal happens. It's not obvious from reading the code, and I think it would be confusing if I wasn't reviewing a change for exactly this reason.